invenio_users_resources/records/systemfields/__init__.py

```diff
@@ -0,0 +1,169 @@
+        elif user_record.preferences["visibility"] == "public":
+            return "profile"
+        else:
+            return "hidden"
+
+
+class IsNotNoneField(CalculatedIndexedField):
+    """Dump a bool for easier checking if a value is set."""
+
+    def __init__(self, field, *args, **kwargs):
+        """Constructor."""
+        self._field = field
+        super().__init__(*args, **kwargs)
+
+    def calculate(self, user_record):
+        """Checks if a timestamp is not none."""
+        return getattr(user_record, self._field, None) is not None
+
+
+class DomainField(CalculatedIndexedField):
+    """Get information about the user's domain."""
+
+    def calculate(self, user_record):
+        """Checks if a timestamp is not none."""
+        domain = current_datastore.find_domain(user_record.domain)
+        if domain is None:
+            return {
+                "tld": "",
+                "status": "1",
+                "category": None,
+                "flagged": False,
+            }
+        else:
+            return {
+                "tld": domain.tld,
+                "status": domain.status.value,
+                "category": domain.category,
+                "flagged": domain.flagged,
+            }
+
+
+class UserIdentitiesField(CalculatedIndexedField):
+    """Get a user's different linked account identities."""
+
+    def calculate(self, user_record):
+        """Checks if a timestamp is not none."""
+        identities = UserIdentity.query.filter_by(id_user=user_record.id).all()
+        data = {}
+        for i in identities:
+            data[i.method] = i.id
+        return data
+
+
+class DomainOrgField(CalculatedIndexedField):
+    """Get information about the user's domain."""
+
+    def calculate(self, domain_record):
+        """Checks if a timestamp is not none."""
+        if not domain_record.model.org_id:
+            return None
+
+        org = domain_record.model.model_obj.org
+
+        parent_org = None
+        if org.parent_id is not None:
+            parent_org = org.parent
+
+        data = [
+            {
+                "id": org.id,
+                "pid": org.pid,
+                "name": org.name,
+                "props": org.json or {},
+                "is_parent": False,
+            }
+        ]
+        if parent_org:
+            data.append(
+                {
+                    "id": parent_org.id,
+                    "pid": parent_org.pid,
+                    "name": parent_org.name,
+                    "props": parent_org.json or {},
+                    "is_parent": True,
+                }
+            )
+        return data
+
+
+class DomainCategoryNameField(CalculatedIndexedField):
+    """Dump the name of the category."""
+
+    def calculate(self, domain_record):
+        """Logic for calculating."""
+        if domain_record.model.model_obj.category:
+            return domain_record.model.model_obj.category_name.label
+        else:
+            return None
```

Comment:
```suggestion
        category = domain_record.model.model_obj.category
        return category.category_name.label if category else None
```

Reply:
I also left this one as is, because ``category_name`` is not a property on the category, but on the ``model_obj`` (ie. ``model_obj.category_name`` instead of ``model_obj.category.category_name``.